packages/request-manager/src/torControlPort.ts

```diff
@@ -123,6 +123,37 @@ export class TorControlPort {
         return this.socket.write(`${command}\r\n`);
     }
 
+    // https://gitweb.torproject.org/torspec.git/tree/control-spec.txt
+    sendCommand(command: string) {
+        return new Promise<TorCommandResponse>(resolve => {
+            this.socket.once('data', data => {
```

Comment:
When the circuit are going to be closed due to misbehaving I see the error below:

```
(node:25074) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 data listeners added to [Socket]. Use emitter.setMaxListeners() to increase limit
```
It probably makes sense since there are probably many circuits and at the moment of closing them there is at least 1 EventEmitter for each.  But not sure if that is the root cause. Anyway probably worth considering.

Reply:
thanks, this should be solved by using `promiseAllSequence` util